src/Halogen/Aff/Driver.purs

```diff
@@ -117,25 +117,28 @@ runUI renderSpec component i = do
   Eval.handleLifecycle lchs do
     sio <- HS.create
     dsx <- Ref.read =<< runComponent lchs (liftEffect <<< HS.notify sio.listener) i component
-    unDriverStateX (\st ->
-      pure
-        { query: evalDriver disposed st.selfRef
-        , messages: sio.emitter
-        , dispose: dispose disposed lchs dsx
-        }) dsx
+    unDriverStateX
+      ( \st ->
+          pure
+            { query: evalDriver disposed st.selfRef
+            , messages: sio.emitter
+            , dispose: dispose disposed lchs dsx
+            }
+      )
+      dsx
 
   where
 
   evalDriver
     :: forall s f' act ps i' o'
      . Ref Boolean
     -> Ref (DriverState r s f' act ps i' o')
-    -> forall a. (f' a -> Aff (Maybe a))
+    -> forall a
+     . (f' a -> Aff (Maybe a))
```

Comment:
This is also a bit strange. Should we parenthesize this whole line?

```purs
-> (forall a. f' a -> Aff (Maybe a))
```

Reply:
If you want it to appear on the same line, you would need to parenthesize it. That's because `forall` is a quantifier like `->`. For example `forall a. forall b. forall c ...` should format as
```
foo
  :: forall a
   . forall b
   . forall c
   . ...
```